### PR TITLE
Handles POST requests

### PR DESCRIPTION
**Pull request by gilroman**

### CODE REVIEW

File: src/main/java/gil/server/JSONDataStore.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package gil.server;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonReader;
+import javax.json.stream.JsonParsingException;
+import java.io.*;
```
Comment (tbaik) [inline src/main/java/gil/server/JSONDataStore.java]: no * imports :)

File: src/main/java/gil/server/JSONDataStore.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import static java.util.stream.Collectors.toList;
+
+
+public class JSONDataStore {
+    private List<Person> people = new ArrayList<>();
```
Comment (tbaik) [inline src/main/java/gil/server/JSONDataStore.java]: Correct me if I'm wrong but isn't it best practice to state what kind of an arraylist you want to create? So this should probably read `new ArrayList<Person>()` no?

Comment (tbaik) [reply src/main/java/gil/server/JSONDataStore.java]: Also, looking at this file, it doesn't seem to live upto its name. It seems to have much more responsibility than just dealing with JSON. It seems like the logic behind dealing with people should not belong in here.

File: src/main/java/gil/server/PostPersonController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+
+                    Integer personId = person.getId();
+                    response.setProtocol("HTTP/1.1");
+                    response.setStatusCode("201");
+                    response.setReasonPhrase("Created");
+                    response.setContentType("application/json");
+                    response.setLocation("Location: /api/people/" + personId);
```
Comment (tbaik) [inline src/main/java/gil/server/PostPersonController.java]: again, Location and Content Type is also just an http header, you dont need a method for each different header

File: src/main/java/gil/server/PostPersonController.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                    try {
+                        person = data.createPerson(name, email);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+
+                    Integer personId = person.getId();
+                    response.setProtocol("HTTP/1.1");
+                    response.setStatusCode("201");
+                    response.setReasonPhrase("Created");
```
Comment (tbaik) [inline src/main/java/gil/server/PostPersonController.java]: I believe you should be able to derive that this is supposed to say "Created" from the status code. You probably dont need to set it each time.

File: src/main/java/gil/server/JSONDataStore.java (hunk 1 above)
Comment (gilroman) [reply src/main/java/gil/server/JSONDataStore.java]: Thanks for catching this : )

File: src/main/java/gil/server/PostPersonController.java (hunk 4 above)
Comment (gilroman) [reply src/main/java/gil/server/PostPersonController.java]: I like this idea.

File: src/main/java/gil/server/PostPersonController.java (hunk 3 above)
Comment (gilroman) [reply src/main/java/gil/server/PostPersonController.java]: Ok. Implemented this on the previous PR for the file-options branch.

File: src/main/java/gil/server/JSONDataStore.java (hunk 2 above)
Comment (gilroman) [reply src/main/java/gil/server/JSONDataStore.java]: I was following IntelliJ's suggestion for the use of the diamond (<>) on line 18, but did a little bit of research and from Java 7 on one can replace type arguments for Generic Classes with
[The Diamond](https://docs.oracle.com/javase/tutorial/java/generics/types.html#diamond).

Agree that JSONDataStore has too much responsibility. I moved the logic dealing with people to a new class called People Data.

**Pull request merged**